react/features/app-settings/components/AbstractAppSettings.js

```diff
@@ -108,14 +109,48 @@ export class AbstractAppSettings extends Component<Props> {
         });
     }
 
+    _onFinalizeServerURL: () => void;
+
+    /**
+     * Cleans up server URL when the user leaves the field or the screen.
+     * This is required since we want to let the user to freely edit the field
+     * without any limitation at edit time.
+     *
+     * @protected
+     * @returns {void}
+     */
+    _onFinalizeServerURL() {
+        const { serverURL } = this.props._profile;
+
+        if (serverURL) {
+            const urlRegExp = new RegExp('^(\\w+(?:s)?://)?(.+)$');
```

Comment:
This still doesn't catch all problems. White space (very common) for example:

```
urlRegExp.exec('h       t   t  p   s://example.com');
Array [ "h       t   t  p   s://example.com", undefined, "h       t   t  p   s://example.com" ]
```

Reply:
Not sure what is expected here. The example you gave (being very weird) is considered a valid URL even by Chrome's strict URL parser:
new URL('https://h t tp://google.hu')
host:"h%20t%20tp"
hostname:"h%20t%20tp"
href:"https://h%20t%20tp//google.hu"
origin:"https://h%20t%20tp"
password:""
pathname:"//google.hu"
port:""
protocol:"https:"

I don't think we can (or should) fix the user's mistake at any cost, for example by trying to guess what he/she meant before mistyping.

the URL object instantiation will catch most problems though, when the URL is polyfilled with a less lenient library - which is not the goal of this task.

Reply:
No, you have used another URL. Try mine (note the space in the scheme):

```
new URL('h  ttps://example.com');
VM732:1 Uncaught TypeError: Failed to construct 'URL': Invalid URL
    at <anonymous>:1:1
```

If spaces are found elsewhere then Chrome percent encodes them, but since URLs cannot contain spaces (https://tools.ietf.org/html/rfc2396#section-2.4.3) we should skip them and not pretend it will work.

We already do this when parsing URLs: https://github.com/jitsi/jitsi-meet/commit/a23eec55e82096a7c91bb873f6cf72106a9af1f9

Come to think of it, we should probably be using https://github.com/jitsi/jitsi-meet/blob/master/react/features/base/util/uri.js#L170 instead of re-inventing anything here.

Reply:
There you go. However parseStandardURIString still seems to be lenient, as https://!/ seems to be a good URL for it. Anyhow, I hope we're a bit closer.